src/main/java/persistence/sql/definition/ColumnDefinition.java

```diff
@@ -0,0 +1,129 @@
+    private final String declaredName;
+    private final boolean nullable;
+    private final int length;
+
+    public ColumnDefinition(Field field) {
+        this.declaredName = field.getName();
+        this.name = determineColumnName(field);
+        this.sqlType = determineColumnType(field);
+        this.nullable = determineColumnNullable(field);
+        this.length = determineColumnLength(field);
+    }
+
+    private static String determineColumnName(Field field) {
+        final String columnName = field.getName();
+
+        if (field.isAnnotationPresent(Column.class)) {
+            Column column = field.getAnnotation(Column.class);
+            if (!column.name().isEmpty()) {
+                return column.name();
+            }
+        }
+
+        return columnName;
+    }
+
+    private static SqlType determineColumnType(Field field) {
+        final String entityFieldType = field.getType().getSimpleName();
+        return SqlType.from(entityFieldType);
+    }
+
+    private static int determineColumnLength(Field field) {
+        if (field.isAnnotationPresent(Column.class)) {
+            Column column = field.getAnnotation(Column.class);
+            return column.length();
+        }
+        return DEFAULT_LENGTH;
+    }
+
+    private static boolean determineColumnNullable(Field field) {
+        final boolean hasColumnAnnotation = field.isAnnotationPresent(Column.class);
+        if (!hasColumnAnnotation) {
+            return true;
+        }
+        return field.getAnnotation(Column.class).nullable();
+    }
+
+    public String name() {
+        return name;
+    }
+
+    public SqlType sqlType() {
+        return sqlType;
+    }
+
+    public boolean shouldNotNull() {
+        return !nullable;
+    }
+
+    public int length() {
+        return length;
+    }
+
+    public boolean nullable() {
+        return nullable;
+    }
+
+    public String declaredName() {
+        return declaredName;
+    }
+
+    public boolean hasValue(Object entity) {
+        final Field[] declaredFields = entity.getClass().getDeclaredFields();
+        final Field targetField = findMatchingField(declaredFields);
+
+        return getValueFromObject(entity, targetField).isPresent();
+    }
+
+    public Object valueAsString(Object entity) {
+        final Field[] declaredFields = entity.getClass().getDeclaredFields();
+        final Field targetField = findMatchingField(declaredFields);
+
+        return getValueFromObject(entity, targetField)
+                .orElseThrow(() -> new NoSuchElementException("Value is null"));
+    }
+
+    private Field findMatchingField(Field[] declaredFields) {
+        for (Field field : declaredFields) {
+            if (field.getName().equals(declaredName())) {
+                return field;
+            }
+        }
+
+        throw new NoSuchElementException("Field not found");
+    }
+
+    @NotNull
+    private static Optional<Object> getValueFromObject(Object entity, Field field) {
```

Comment:
```suggestion
    private Optional<Object> getValueFromObject(Object entity, Field field) {
```
👀 intellij extract 단축에서 static이 들어갈 수 있어서 유의하시면 좋을 것 같아요 :)